docs/technical/pull-requests.md

```diff
@@ -0,0 +1,44 @@
+# Checking out pull requests from a forked repository
+PR [#3602](https://github.com/desktop/desktop/pull/3602) introduced the ability to checkout a branch from a forked repository. In order to accomplish this, we needed a way to manage remotes on your behalf. This document is intended to detail the process we developed to make checking out PRs as frictionless as possible.
+
+## Magic Remote Prefix
+One of the main problems we needed to solve was determining which remotes are no longer needed and can be cleaned. We decided to prefix the remotes we add on your behalf with a magic string: `github-desktop-`
+
+```ts
+export const ForkedRemotePrefix = 'github-desktop-'
+```
+[Code](https://github.com/desktop/desktop/blob/34a05b155ff69bb19cc4da5b2caa89856e3e63fb/app/src/lib/stores/pull-request-store.ts#L26)
+
+## Cleaning Remotes
```

Comment:
I think this section should be moved up, since it's defining the problem that we're trying to solve with the magic prefix.